src/test/java/org/mule/mvel2/tests/core/TypesAndInferenceTests.java

```
@@ -1670,6 +1671,25 @@ public void testCompiledAccExpressionSerialization() throws IOException, ClassNo
     assertNull(readCompiledAccExpression.getAccessor());
   }
 
+  public void testInputTypeChange() {
```

Isnt the fix just for the case when a failure is ignored because of COMPILER_OPT_PROPERTY_ACCESS_DOESNT_FAIL?